Sources/xcodes/main.swift

```diff
@@ -21,11 +26,41 @@ func loginIfNeeded() -> Promise<Void> {
     }
     .recover { error -> Promise<Void> in
         guard
-            let username = env("XCODES_USERNAME") ?? readLine(prompt: "Apple ID: "),
-            let password = env("XCODES_PASSWORD") ?? readSecureLine(prompt: "Apple ID Password: ")
+            let username = findUsername() ?? readLine(prompt: "Apple ID: "),
+            let password = findPassword() ?? readSecureLine(prompt: "Apple ID Password: ")
         else { throw Error.missingUsernameOrPassword }
 
-        return manager.client.login(accountName: username, password: password)
+        return login(username, password: password)
+    }
+}
+
+func findUsername() -> String? {
+    if let username = env(xcodesUsername) {
+        return username
+    }
+    else if let username = try? keychain.getString(xcodesUsername){
+        return username
+    }
+    return nil
+}
+
+func findPassword() -> String? {
+    if let password = env(xcodesPassword) {
+        return password
+    }
+    else if let password = try? keychain.getString(xcodesPassword){
```

Comment:
This (and the equivalent code for the username) should use the same key as is used to set the username and password in the Keychain after a successful login, right?

Reply:
As per my above comment `keychain["xcodesUsername"]` was wrong in setting it, so yes they should be using the same key